tests/test_isrFunctions.py

```diff
@@ -322,6 +301,38 @@ def test_attachTransmissionCurve_None(self):
         # DM-19707: ip_isr functionality not fully tested by unit tests
         self.assertIsNotNone(combined)
 
+    def test_countMaskedPixels(self):
+        mockImageConfig = isrMock.IsrMock.ConfigClass()
+
+        # flatDrop is not really relevant as we replace the data
+        # but good to note it in case we change how this image is made
+        mockImageConfig.flatDrop = 0.99999
+        mockImageConfig.isTrimmed = True
+
+        flatExp = isrMock.FlatMock(config=mockImageConfig).run()
+        (shapeY, shapeX) = flatExp.getDimensions()
+
+        rng = np.random.RandomState(0)
+        flatMean = 1000
+        flatWidth = np.sqrt(flatMean)
+        flatData = rng.normal(flatMean, flatWidth, (shapeX, shapeY))
+        flatExp.image.array[:] = flatData
+
+        exp = flatExp.clone()
+        mi = exp.maskedImage
+        self.assertEqual(ipIsr.countMaskedPixels(mi, 'NO_DATA'), 0)
+        self.assertEqual(ipIsr.countMaskedPixels(mi, 'BAD'), 0)
+
+        NODATABIT = mi.mask.getPlaneBitMask("NO_DATA")
+        noDataBox = geom.Box2I(geom.Point2I(31, 49), geom.Extent2I(3, 6))
+        mi.mask[noDataBox] |= NODATABIT
+
+        self.assertEqual(ipIsr.countMaskedPixels(mi, 'NO_DATA'), noDataBox.getArea())
+        self.assertEqual(ipIsr.countMaskedPixels(mi, 'BAD'), 0)
+
+        mi.mask[noDataBox] ^= NODATABIT  # XOR to reset what we did
+        self.assertEqual(ipIsr.countMaskedPixels(mi, 'NO_DATA'), 0)
```

Comment:
What exactly is happening in this unit test? I see all comparisons to zero. Are these null tests always? Shouldn't there be tests for when the number of masked pixels is different from zero? Maybe I misunderstood the test.

Reply:
The test above this line applies a mask bit (NO_DATA) to a region, and then checks that the count of those pixels in that mask is equal to the area of that region.  The check for BAD is checking that it's correctly respecting mask planes and only measuring what it's told.